### DISCUSSION TITLE
Paginator: Add ability to provide custom elements for paginatorTemplate

### DISCUSSION THREAD

**Discussion by ikasianiuk** [primefaces/community#2041] [PrimeVue] [answered]
hi, first of all thanks for the great library and all the work you are doing. 

I have a custom design for a pagination which looks like this 
<img width="543" alt="image" src="https://github.com/primefaces/community/assets/24921938/479f1c6b-9d0b-491b-9eda-341037e634bb">
It is actually quite popular and can be seen on lots of websites.

As far as I can see currently there is no way to make paginator look like on the picture above. Though I can see that in PrimeReact there is possibility [template prop](https://primereact.org/paginator/#template) to pass custom elements through template prop.

Can this also be implemented for PrimeVue?

**Comment by tugcekucukoglu** [comment, marked as answer]
Hi,

Also PrimeVue Paginator has same slots: start, end, firsticon, lasticon, etc.
https://v4.primevue.org/paginator/#api.paginator.slots

**Reply by ikasianiuk** [reply]
hi @tugcekucukoglu, thanks for your response. 
Maybe I wasn't quite clear. I saw these slots and I understand that i can set custom icons using them and add some content on the left and right sides of the paginator. But what I actually need is the ability to add these dots between the pages, like on the first example here https://primereact.org/paginator/#template
<img width="816" alt="image" src="https://github.com/primefaces/community/assets/24921938/75d238fa-4ee5-4fc9-a206-70e55ab3f05c">
 
Currently these slots doesn't provide the possibility to add this dots or am I missing anything?

**Reply by RebeccaStevens** [reply]
I would also like the option to use a custom button element.

**Comment by urzhanrisbek99** [comment]
Hello, any updates on that?
